ironfish/src/wallet/account/encoder/bech32.ts

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { PUBLIC_ADDRESS_LENGTH } from '@ironfish/rust-nodejs'
+import bufio from 'bufio'
+import { Bech32m } from '../../../utils'
+import {
+  AccountImport,
+  KEY_LENGTH,
+  VERSION_LENGTH,
+  VIEW_KEY_LENGTH,
+} from '../../walletdb/accountValue'
+import { AccountEncoder } from './encoder'
+
+export const BECH32_ACCOUNT_PREFIX = 'ironfishaccount00000'
+
+export class Bech32AccountEncoder implements AccountEncoder {
+  encode(value: AccountImport): string {
+    const bw = bufio.write(this.getSize(value))
+    bw.writeU16(value.version)
+
+    let flags = 0
+    flags |= Number(!!value.spendingKey) << 0
+    flags |= Number(!!value.createdAt) << 1
+    bw.writeU8(flags)
```

Comment:
I'm considering a new way of optional behaviour which is just using a byte at the beginning of each item. I feel like we constantly run out of space in these flags. It's also easier to just see right before something is serialized, a 1 byte read. What do you think? It'll use extra bytes for each optional thing, but it might make our serialization code cleaner.

Reply:
For optional stuff we would only need 1 bit, not a byte, but I like this idea. I thought the bitflags were kind of fragile.